cli/commands/list/list.go

```diff
@@ -209,7 +233,19 @@ func discoveredToListed(components component.Components, opts *Options) (ListedC
 		listedCfg.Dependencies = make([]*ListedComponent, len(c.Dependencies()))
 
 		for i, dep := range c.Dependencies() {
-			relDepPath, err := filepath.Rel(opts.WorkingDir, dep.Path())
+			// Ensure dependency path is absolute for reliable relativity calculation
+			depPath := dep.Path()
+			if !filepath.IsAbs(depPath) {
+				var err error
+
+				depPath, err = filepath.Abs(depPath)
```

Comment:
I think this is wrong. We never want to care about the PWD of the process. We need the path to always be relative to the --working-dir flag.